roachtest: sqlsmith/setup=empty/setting=default failed

roachtest.sqlsmith/setup=empty/setting=default [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3305136&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3305136&tab=artifacts#/sqlsmith/setup=empty/setting=default) on master @ [04a41e7915f4a89dcc1d0dbd92466c6adf79ec9f](https://github.com/cockroachdb/cockroach/commits/04a41e7915f4a89dcc1d0dbd92466c6adf79ec9f):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/sqlsmith/setup=empty/setting=default/run_1
	sqlsmith.go:217,sqlsmith.go:252,test_runner.go:777: error: pq: internal error: runtime error: index out of range [-1]
		stmt:
		SELECT
			tab_632.col0_5 AS col_1039,
			tab_633.col0_1 AS col_1040,
			(-1.3167076110839844):::FLOAT8 AS col_1041,
			tab_631.col0_2 AS col_1042
		FROM
			defaultdb.public.tab_513@tab_513_col0_5_col0_4_col0_1_col0_3_col0_0_idx AS tab_631
			JOIN defaultdb.public.tab_513@tab_513_col0_5_col0_4_col0_1_col0_3_col0_0_idx AS tab_632 ON
					(tab_631.col0_5) = (tab_632.col0_5),
			defaultdb.public.tab_513@[0] AS tab_633;
```
<details><summary>Reproduce</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/roachtest)



See: [CI job to stress roachtests](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestStress)

<p>For the CI stress job, click the ellipsis (...) next to the Run button and fill in:
* Changes / Build branch: master
* Parameters / `env.TESTS`: `^sqlsmith/setup=empty/setting=default$`
* Parameters / `env.COUNT`: &lt;number of runs&gt;
</p>
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=empty/setting=default.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/opt/xform/join_funcs.go
+++ b/pkg/sql/opt/xform/join_funcs.go
@@ -1181,7 +1181,7 @@ func (c *CustomFuncs) findJoinFilterRange(
 ) (filterIdx int, ok bool) {
 	for filterIdx := range filters {
 		props := filters[filterIdx].ScalarProps()
-		if props.TightConstraints && !props.Constraints.IsUnconstrained() {
+		if props.TightConstraints && props.Constraints.Length() > 0 {
 			constraint := props.Constraints.Constraint(0)
 			constraintCol := constraint.Columns.Get(0).ID()
 			// See comment in findFiltersForIndexLookup for why we check filter here.

--- a/pkg/sql/opt/xform/testdata/rules/join
+++ b/pkg/sql/opt/xform/testdata/rules/join
@@ -9849,3 +9849,36 @@ left-join (lookup metric_values)
  │    └── fd: (1)-->(2)
  └── filters
       └── name:2 = 'cpu' [outer=(2), constraints=(/2: [/'cpu' - /'cpu']; tight), fd=()-->(2)]
+
+# Regression test for #68975. Ensure that findJoinFilterRange does not panic
+# from trying to access the 0-th constraint in an empty constraint set.
+exec-ddl
+CREATE TABLE t68975 (
+    s STRING,
+    i INT CHECK (false),
+    INDEX (s DESC)
+);
+----
+
+opt
+SELECT NULL
+FROM t68975 AS t1 JOIN t68975 AS t2
+ON t1.s = t2.s;
+----
+project
+ ├── columns: "?column?":11
+ ├── fd: ()-->(11)
+ ├── inner-join (merge)
+ │    ├── columns: t1.s:1!null t2.s:6!null
+ │    ├── left ordering: -1
+ │    ├── right ordering: -6
+ │    ├── fd: (1)==(6), (6)==(1)
+ │    ├── scan t68975@secondary [as=t1]
+ │    │    ├── columns: t1.s:1
+ │    │    └── ordering: -1
+ │    ├── scan t68975@secondary [as=t2]
+ │    │    ├── columns: t2.s:6
+ │    │    └── ordering: -6
+ │    └── filters (true)
+ └── projections
+      └── NULL [as="?column?":11]

